controllers/admin/base_admin_controller.js

```diff
@@ -0,0 +1,75 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module.exports = function BaseAdminControllerModule(pb) {
+  "use strict";
+
+  var util = pb.util;
+  var BaseController = pb.BaseController;
+
+  /**
+   * This class serves as a base class for all the controllers used in the admin control panel
+   * @constructor
+   */
+  function BaseAdminController() {}
+  util.inherits(BaseAdminController, BaseController);
+
+  /**
+   * Initializes the admin controller with site-related info
+   * @override
+   * @param props
+   * @param cb
+   */
+  BaseAdminController.prototype.init = function (props, cb) {
+    var self = this;
+    BaseController.prototype.init.call(self, props, function () {
+      self.extendedInit(cb);
+    });
+  };
+
+  BaseAdminController.prototype.extendedInit = function(cb) {
+    this.siteQueryService = new pb.SiteQueryService(this.site, true);
+    this.settings = pb.SettingServiceFactory.getServiceBySite(this.site, true);
+    cb();
+  };
+
+  /**
+   * Retrieves a context object that contains the necessary information for
+   * service prototypes
+   * @method getServiceContext
+   * @return {Object}
+   */
+  BaseAdminController.prototype.getServiceContext = function(){
+    return util.merge(BaseAdminController.super_.prototype.getServiceContext.apply(this), { onlyThisSite: true});
+  };
```

Comment:
I have updated this with your suggestion.

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, July 9, 2015 at 5:21 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In controllers/admin/base_admin_controller.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34305112>:

> +
> +  BaseAdminController.prototype.extendedInit = function(cb) {
> +    this.siteQueryService = new pb.SiteQueryService(this.site, true);
> +    this.settings = pb.SettingServiceFactory.getServiceBySite(this.site, true);
> +    cb();
> +  };
> +
> +  /**
> +   * Retrieves a context object that contains the necessary information for
> +   * service prototypes
> +   * @method getServiceContext
> +   * @return {Object}
> +   */
> +  BaseAdminController.prototype.getServiceContext = function(){
> +    return util.merge(BaseAdminController.super_.prototype.getServiceContext.apply(this), { onlyThisSite: true});
> +  };


The base controller is already creating a new context object each time getServiceContext is called. It is not necessary to call merge.

var context = BaseAdminController.super_.prototype.getServiceContext.apply(this);
context.onlyThisSite = true;
return context;


-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34305112>.